charmstore_internal_test.go

```diff
@@ -177,12 +177,13 @@ func (ResourceSuite) TestPublish(c *gc.C) {
 		"data": 1,
 		"lib":  2,
 	}
-	result, err := s.Publish(id, resources)
+	result, err := s.Publish(id, []string{"development"}, resources)
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(result, gc.DeepEquals, idrev)
 	f.CheckCall(c, 0, "PutWithResponse", "/"+id.Path()+"/publish", &params.PublishRequest{
```

Comment:
Added a full stack test that publishes to development then tries to get from dev and stable (obv checking that it fails when retrieving from stable).